### ISSUE TITLE
Drummer's web clips bookmarklet

### ISSUE THREAD

**Issue by scripting** [scripting/drummerSupport#96]
A new feature today, [Web clips & the bookmarklet](http://docserver.scripting.com/drummer/general.opml#1634950078000).

This is a place for comments and questions.

**Comment by dmatchett** [comment]
@scripting -- looks like you've solved all the bookmarklet issues -- thanks and yay! There are quirks and different behaviors in different circumstances -- maybe I'll blog about those, but it keeps on working so that's only local color. My favorite drummer feature so far -- I'm using it daily with or without any visible blog posts. As previously mentioned: this is the first bookmark manager that lets me arrange things into directories, kind of an old-style personal Yahoo.

**Comment by scripting** [comment]
@dmatchett — what an interesting concept, a personal yahoo.

**Comment by dmatchett** [comment]
I spoke too soon. I was on the <a href="http://scripting.com/2015/05/13/howToHaveAFuture.html">"May 2015"</a> page from this morning's Scripting News. I clicked the bookmarklet. My browser was open to my "Bookmarks" tab, not the "Notes" tab. In a testing mood, I closed all tabs at that point and reloaded. It reopened to the Bookmarks tab. There was no link showing on the Notes tab. I reloaded -- no joy. I left the Notes tab showing, returned to a different Safari tab and re-clicked the bookmarklet for "May 2015" aka <a href="http://scripting.com/2015/05/13/howToHaveAFuture.html">"How to have a Future."</a> Again, nothing showed, but when I hit Reload (i.e. command-R) the link appeared. That's good enough for me, but I think the glitch deserves tracking, just low priority. I'm content with the overall.

**Comment by scripting** [comment]
I'm having trouble parsing this. 

Can you summarize -- what was wrong?

**Comment by dmatchett** [comment]
Will try -- requires some form of attention -- poets, you know... :)

**Comment by jackbaty** [comment]
The bookmarklet itself seems to work well for me (tested today). A new (Safari) tab is opened and a new entry is created in the correct place (date) in the Notes outline.

The unexpected behavior I'm seeing is an unwanted redirect each time I use the bookmarklet.

I keep Drummer open in a "pinned" Safari tab. After I use the Drummer bookmarklet, that tab is automatically redirected to http://docserver.scripting.com/drummer/about.opml. Using the back button doesn't help, as the redirect happens again. The only way out of it is to manually type "drummer.scripting.com" in that pinned tab. (It does not matter whether the tab is pinned or not). It's very strange.

Here's a video that I hope demonstrates the phenomenon:

https://user-images.githubusercontent.com/970/142024545-a525987b-71d9-4361-953e-2d134c83758d.mp4

In the video I:

1. Highlighted some text on a web page
2. Clicked the Drummer bookmarklet
3. Showed that the entry was created correctly in Notes.opml in a new browser tab
4. Clicked my pinned Drummer tab, on which you can see that it's been redirected to About Drummer
5. Tried the Back button several times, only to be redirected to About Drummer each time

This may very well be something environmental on my Mac, but I can't figure out what that might be.

**Comment by mistersugar** [comment]
@jackbaty  @troutcolor This relates to an expected behavior, I believe, in which Drummer can only be open in one browser tab and if a new instance is opened the previous tab goes to the Docs page. I will look for previous issue that documented this.

**Comment by scripting** [comment]
@jackbaty -- it has to be that way. i agree -- i wish there were some way to tell the OS tho not reload the app if there's already an instance open. but there is no such thing.

@mistersugar -- yes, Drummer can tell if another instance is opened. the OS will not let it close itself (which would be the rational thing to do, so it redirects (which it will let it do). 

there's a lot of ad hoc behavior in the intersection between the OS and the web.

**Comment by jackbaty** [comment]
I see, and makes sense. Thanks for the clarification, I didn't see any other mentions of it.

I suppose my wish would be that if I close all other Drummer tabs, it could know that, and the back button would then "stick" in my pinned tab. This is certainly work-aroundable, though.

**Comment by scripting** [comment]
A new detailed error report for the bookmarklet. 

https://github.com/scripting/drummerSupport/issues/145
